integreat_cms/core/management/commands/bulk_replace_page_icons.py

```diff
@@ -0,0 +1,145 @@
+
+    # ruff: noqa: S608
+    with connection.cursor() as cursor:
+        cursor.execute(f"""
+            WITH RECURSIVE directory_tree(id, name, parent_id, full_path, is_directory) AS (
+                -- Base case: Select root directories (parent_id IS NULL)
+                SELECT id, name, parent_id, name::TEXT AS full_path, TRUE
+                FROM {directory_table}
+                WHERE parent_id IS NULL
+
+                UNION ALL
+
+                -- Recursive case: Join directories and their subdirectories (children)
+                SELECT d.id, d.name, d.parent_id, (dt.full_path || '/' || d.name)::TEXT, TRUE
+                FROM {directory_table} d
+                JOIN directory_tree dt ON d.parent_id = dt.id
+            ),
+            mediafile_tree AS (
+                -- MediaFile case: Attach MediaFile objects to the full path of the parent directory
+                SELECT mf.id, mf.name, mf.parent_directory_id AS parent_id, (dt.full_path || '/' || mf.name)::TEXT AS full_path, FALSE AS is_directory
+                FROM {mediafile_table} mf
+                JOIN directory_tree dt ON mf.parent_directory_id = dt.id
+            )
+
+            -- Combine the results from both directories and media files
+            SELECT id, full_path, is_directory FROM directory_tree
+            UNION ALL
+            SELECT id, full_path, is_directory FROM mediafile_tree;
+        """)
+
+        rows = cursor.fetchall()
+        return {row[1]: {"id": row[0], "is_directory": row[2]} for row in rows}
+
+
+class Command(LogCommand):
+    """
+    Management command to change references of media files from a source path to a target path.
+    This command reads a CSV file and then changes the reference from the first column (source path) to the second column (target path).
+    """
+
+    help = "Replace references of media files from source paths with target paths based on a CSV file."
+
+    def add_arguments(self, parser: CommandParser) -> None:
+        """
+        Define the arguments of this command
+
+        :param parser: The argument parser
+        """
+        parser.add_argument(
+            "path",
+            nargs="?",
+            default=None,
+            help="The path to the csv file we want to import from",
+        )
+
+    def should_stdin_be_used(self, options: dict) -> bool:
+        return bool(
+            sys.platform != "win32"
+            and not sys.stdin.isatty()
+            and select.select([sys.stdin], [], [], 0)[0]
+            and not options["path"]
+        )
+
+    def handle(self, *args: Any, **options: Any) -> None:
+        successful = 0
+        failed = 0
+
+        with (
+            open(options["path"], newline="")
+            if not self.should_stdin_be_used(options)
+            else sys.stdin as csvfile
+        ):
+            reader = csv.reader(csvfile, delimiter=",", quotechar="|")
+            next(reader, None)
+            full_path_lookup = get_mixed_tree_paths()
```

Comment:
Is this variable used anywhere?

Reply:
Sorry, we didn't end up using it as we found it not useful for what the script needs to be after all. I didn't want to throw `get_mixed_tree_paths()` away and forget about it, so I left it in for now. I want to migrate it to somewhere where it makes sense before merging this.